invenio_search/utils.py

```diff
@@ -9,52 +9,108 @@
 """Utility functions for search engine."""
 
 import os
+import time
 
+import six
 from flask import current_app
 
+from .proxies import current_search, current_search_client
 
-def prefix_index(app, index):
+
+def timestamp_suffix():
+    """Generate a suffix based on the current time."""
+    return '-' + str(int(time.time()))
+
+
+def prefix_index(index, prefix=None, app=None):
     """Prefixes the given index if needed.
 
-    :param app: Flask app to get the config from.
     :param index: Name of the index to prefix.
+    :param prefix: Force a prefix.
+    :param app: Flask app to get the prefix config from.
     :returns: A string with the new index name prefixed if needed.
     """
-    index_prefix = app.config['SEARCH_INDEX_PREFIX'] or ''
+    app = app or current_app
+    index_prefix = (prefix if prefix is not None
+                    else (app.config.get('SEARCH_INDEX_PREFIX')) or '')
     return index_prefix + index
 
 
-def build_index_name(app, *parts):
+def suffix_index(index, suffix=None, app=None):
+    """Suffixes the given index.
+
+    :param index: Name of the index to prefix.
+    :param suffix: The suffix to append to the index name.
+    :param app: Flask app to get the "invenio-search" extension from.
+    :returns: A string with the new index name suffixed.
+    """
+    search_ext = app.extensions['invenio-search'] if app else current_search
+    suffix = suffix if suffix is not None else search_ext.current_suffix
+    return index + suffix
+
+
+def build_index_from_parts(*parts):
     """Build an index name from parts.
 
-    :param parts: Parts that should be combined to make an index name.
+    :param parts: String values that will be joined by dashes ("-").
     """
-    base_index = os.path.splitext(
-        '-'.join([part for part in parts if part])
-    )[0]
+    return '-'.join([part for part in parts if part])
 
-    return prefix_index(app=app, index=base_index)
+
+def build_alias_name(index, prefix=None, app=None):
+    """Build an alias name.
+
+    :param index: Name of the index.
+    :param prefix: The prefix to prepend to the index name.
+    """
+    return build_index_name(index, prefix=prefix, suffix='', app=app)
+
+
+def build_index_name(index, prefix=None, suffix=None, app=None):
+    """Build an index name.
+
+    :param index: Name of the index.
+    :param prefix: The prefix to prepend to the index name.
+    :param suffix: The suffix to append to the index name.
+    :param app: Flask app passed to ``prefix_index`` and ``suffix_index``.
+    """
+    if not isinstance(index, six.string_types):
+        index = build_index_from_parts(*index)
+    index = prefix_index(index, prefix=prefix, app=app)
+    index = suffix_index(index, suffix=suffix, app=app)
+    return index
 
 
-def schema_to_index(schema, index_names=None):
+def schema_to_index(schema, index_names=None, prefixed=True):
```

Comment:
This can be just `prefix=None` and be passed directly to the `build_index_name` below (see comment below as well though)